### PR TITLE
feature amazon resourceExists

### PR DESCRIPTION
**Pull request by ben-vr**

### CODE REVIEW

File: src/lib/amazon.ts [hunk 1]
```diff
@@ -70,6 +70,22 @@ export async function getPresignedUrl(client: S3, options: IPresignedS3Options):
   }
 }
 
+/**
+ * Checks if the resource exists on the S3 bucket
+ * @param {Object} client existing s3 client
+ * @param {Object} options s3 options
+ * @returns {boolean} resource exists or not
+ */
+export async function resourceExists(client: S3, params: {bucket: string, key: string}): Promise<boolean> {
```
Comment (horstenwillem) [inline src/lib/amazon.ts]: Add tests please

File: src/lib/amazon.ts [hunk 2]
```diff
@@ -70,6 +70,22 @@ export async function getPresignedUrl(client: S3, options: IPresignedS3Options):
...
+  const { bucket, key } = params;
+  return new Promise((resolve, _reject) => {
+    client.headObject({ Bucket: bucket, Key: key }, (err, _data) => {
+      if (err)  resolve(false);
```
Comment (horstenwillem) [inline src/lib/amazon.ts]: Double spaces, turn on lintr?

File: src/lib/amazon.ts [hunk 3]
```diff
@@ -70,6 +70,22 @@ export async function getPresignedUrl(client: S3, options: IPresignedS3Options):
...
+ * Checks if the resource exists on the S3 bucket
+ * @param {Object} client existing s3 client
+ * @param {Object} options s3 options
+ * @returns {boolean} resource exists or not
+ */
+export async function resourceExists(client: S3, params: {bucket: string, key: string}): Promise<boolean> {
+  const { bucket, key } = params;
+  return new Promise((resolve, _reject) => {
+    client.headObject({ Bucket: bucket, Key: key }, (err, _data) => {
+      if (err) resolve(false);
```
Comment (knor-el-snor) [inline src/lib/amazon.ts]: Use `return` for cleaner code> Added Stackoverflow article which has some reasons why
https://stackoverflow.com/questions/32536049/do-i-need-to-return-after-early-resolve-reject/32536083

File: src/lib/amazon.ts [hunk 4]
```diff
@@ -70,6 +70,22 @@ export async function getPresignedUrl(client: S3, options: IPresignedS3Options):
...
+      resolve(true);
```
Comment (knor-el-snor) [inline src/lib/amazon.ts]: Same

File: tests/lib/amazon.test.ts [hunk 5]
```diff
@@ -95,4 +96,11 @@ describe('amazon', () => {
       }
     });
   });
+
+  describe('resourceExists', () => {
+    it('Should return true when the resource exists', async () => {
```
Comment (knor-el-snor) [inline tests/lib/amazon.test.ts]: Also test false promise returning `false`

**Pull request merged**